### PR TITLE
.Net [OpenApi] Converting OpenApi plugins result to string

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
This change is necessary to align the handling of OpenAPI plugin results with the current approach for handling results from native plugins. In this approach, a plugin result is converted to a string to be saved to SKContext if such a conversion is possible. This conversion enables planners to use the result of the first plugin as input for the next one because the result of the first plugin is a string rather than a complex type like an object or JSON.

### Description
- The type of the `RestApiOperationResponse.Content` property has been changed from a string to an object to represent different content types such as JSON, XML, text, images, etc.
- The `RestApiOperationRunner` class has been modified to handle and serialize different content types differently. For example, JSON, text, and XML content are read and serialized using the `HttpContent.ReadAsStringAsync` method, while images are read and serialized using the `HttpContent.ReadAsByteArrayAsync` method.
- A new `RestApiOperationResponseConverter` class has been added to convert the `RestApiOperationResponse` to a string so that it can be used by any other plugin in the pipeline.
- A small regression in the `Plan` class has been fixed. This regression caused a failure in the `StepwisePlanner` if a plugin returned a non-string result.

Fixes the issue - https://github.com/microsoft/semantic-kernel/issues/2280

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Functions/Functions.UnitTests/OpenAPI/RestApiOperationResponseConverterTests.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public RestApiOperationResponseConverterTests()
+    {
+        this._sut = new RestApiOperationResponseConverter();
+    }
+
+    [Fact]
+    public void ItShouldConvertStringContentToString()
+    {
+        //Arrange
+        var response = new RestApiOperationResponse("fake-content", "fake-content-type");
```
Comment (markwallace-microsoft) [inline dotnet/src/Functions/Functions.UnitTests/OpenAPI/RestApiOperationResponseConverterTests.cs:20]: Should the content type be `text/plain` here?

**Review by markwallace-microsoft — COMMENTED** [review]
I was expecting to see an integration test which reproduces the current failure that now passes. Can this be added?

File: dotnet/src/Functions/Functions.UnitTests/OpenAPI/RestApiOperationResponseConverterTests.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/src/Functions/Functions.UnitTests/OpenAPI/RestApiOperationResponseConverterTests.cs:20]: The content type does not matter with the current implementation of the converter, which uses the type of the value of the `Content` property to decide on a conversion mechanism, as opposed to doing so based on the content type. However, the converter implementation can and will be changed to perform conversion based on the content type when and if needed.

**Comment by SergeyMenshykh** [comment]
> I was expecting to see an integration test which reproduces the current failure that now passes. Can this be added?
Yep, make sense, done.

File: dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs [hunk 2]
```diff
@@ -29,6 +29,17 @@ internal sealed class RestApiOperationRunner
...
 
+    /// <summary>
+    /// A dictionary containing the content type as the key and the corresponding content serializer as the value.
+    /// </summary>
+    private static readonly Dictionary<string, HttpResponseContentSerializer> s_serializerByContentType = new()
+    {
+        { "image", async (content) => await content.ReadAsByteArrayAndTranslateExceptionAsync().ConfigureAwait(false) },
+        { "text", async (content) => await content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false) },
+        { "application/json", async (content) => await content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false)},
+        { "application/xml", async (content) => await content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false)}
```
Comment (dmytrostruk) [inline dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs:40]: nit: Looks like there is no serialization difference between `application/json` and `application/xml`. There is also logic which fetches primary content type `image`, `text`, `application` and tries to get serializer by this type. Wondering, if we can leave just `application` primary content type in this dictionary.

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs:40]: Yeah, I initially considered registering a single serialization method using only the primary content-type 'application'. However, I later changed my mind and decided to register only the known ones for now, as there are many different 'application' subtypes for which it is not clear how to serialize - https://www.iana.org/assignments/media-types/media-types.xhtml#application."

Comment (dmytrostruk) [reply dotnet/src/Functions/Functions.OpenAPI/RestApiOperationRunner.cs:40]: Makes sense, thank you!

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**